### PR TITLE
RF01 TC 019 Verificar Búsqueda de Personal

### PR DESCRIPTION
**Pull request by franzelunico**
image
image

### CODE REVIEW

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 1]
```diff
@@ -4,13 +4,17 @@
 import org.openqa.selenium.support.FindBy;
 import org.umssdiplo.automationv01.core.managepage.BasePage;
 import org.umssdiplo.automationv01.core.managepage.Usuario.ListUser;
+import org.umssdiplo.automationv01.core.managepage.Personal.FindPersonal;
 import org.umssdiplo.automationv01.core.utils.CommonEvents;
 
 public class SubMenuPersonal extends BasePage {
 
     @FindBy(xpath = "//button[@id='users']")
     private WebElement usuarioSubMenu;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: Change name WebElement. Ej.: subMenuUsuario

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 2]
```diff
@@ -4,13 +4,17 @@
...
 
+    @FindBy(xpath = "//button[@id='personal']")
+    private WebElement personalSubMenu;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: Change name WebElement. Ej.: subMenuPersonal

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.umssdiplo.automationv01.core.managepage.BasePage;
+import org.umssdiplo.automationv01.core.utils.CommonEvents;
+
+public class FindPersonal extends BasePage {
+
+    @FindBy(name = "buscarPersonal")
+    private WebElement itBuscarPersonal;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java]: Change name WebElement. Ej.: inputFindPerson

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    @FindBy(css = ".mat-paginator-range-label")
+    private WebElement paginator;
+
+    public FindPersonal(){
+        CommonEvents.isVisible(itBuscarPersonal);
+        CommonEvents.isVisible(paginator);
+    }
+
+    public boolean isFindPersonalVisible(){
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java]: change name method.: Ej.: validateInputFindPersonIsVisible()

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        return CommonEvents.isVisible(itBuscarPersonal);
+    }
+
+    public void buscarPersonal(String personal){
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java]: try to use everithing in english. Ej.: setTextFindPerson(String person)

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        CommonEvents.setInputField(itBuscarPersonal, personal);
+    }
+
+    public int personalEncontrado(int x){
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java]: try to use everything in english Ej.: validatePersonFound()

File: src/test/resources/features/Personal/Personal.feature [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+Feature: Personal
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario Outline: Verificar Busqueda de Personal
+    Given 'Menu principal' estee cargado
+    And seleccionar menu 'Personal' en la pagina 'Menu Principal'
+    And Seleccionar submenu 'Personal' en menu 'Personal'
```
Comment (jdpapiero) [inline src/test/resources/features/Personal/Personal.feature]: the first letter in step description should be in lowercase

File: src/test/resources/features/Personal/Personal.feature [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    When Ingresar <correo> para 'Buscar Personal' en la tabla
```
Comment (jdpapiero) [inline src/test/resources/features/Personal/Personal.feature]: the first letter in step description should be in lowercase
is necessary describe better this step

File: src/test/resources/features/Personal/Personal.feature [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    Then El resultado de 'Buscar Personal' deberia ser <resultado>
```
Comment (jdpapiero) [inline src/test/resources/features/Personal/Personal.feature]: the first letter in step description should be in lowercase

**Review by jdpapiero — CHANGES_REQUESTED** [review]
review comments

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 10]
```diff
@@ -4,13 +4,17 @@
...
 import org.umssdiplo.automationv01.core.managepage.Usuario.ListUser;
+import org.umssdiplo.automationv01.core.managepage.Personal.FindPersonal;
 import org.umssdiplo.automationv01.core.utils.CommonEvents;
 
 public class SubMenuPersonal extends BasePage {
 
     @FindBy(xpath = "//button[@id='users']")
     private WebElement usuarioSubMenu;
 
+    @FindBy(xpath = "//button[@id='personal']")
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: If you use id for identify web element, you don't need use 'xpath', please use 'id'

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 11]
```diff
@@ -4,13 +4,17 @@
 import org.openqa.selenium.support.FindBy;
 import org.umssdiplo.automationv01.core.managepage.BasePage;
 import org.umssdiplo.automationv01.core.managepage.Usuario.ListUser;
+import org.umssdiplo.automationv01.core.managepage.Personal.FindPersonal;
 import org.umssdiplo.automationv01.core.utils.CommonEvents;
 
 public class SubMenuPersonal extends BasePage {
 
     @FindBy(xpath = "//button[@id='users']")
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: If you use id for identify web element, you don't need use 'xpath', please use 'id'

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+import org.openqa.selenium.support.FindBy;
+import org.umssdiplo.automationv01.core.managepage.BasePage;
+import org.umssdiplo.automationv01.core.utils.CommonEvents;
+
+public class FindPersonal extends BasePage {
+
+    @FindBy(name = "buscarPersonal")
+    private WebElement itBuscarPersonal;
+
+    @FindBy(css = ".mat-paginator-range-label")
```
Comment (marbusjim) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Personal/FindPersonal.java]: Are you sure this filter it's unique in your page?

File: src/test/resources/features/Personal/Personal.feature [hunk 13]
```diff
@@ -0,0 +1,17 @@
+
+Feature: Personal
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario Outline: Verificar Busqueda de Personal
+    Given 'Menu principal' estee cargado
```
Comment (marbusjim) [inline src/test/resources/features/Personal/Personal.feature]: I think the word 'estee' it's wrong

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 14]
```diff
@@ -4,19 +4,28 @@
...
 import org.umssdiplo.automationv01.core.managepage.Usuario.ListUser;
+import org.umssdiplo.automationv01.core.managepage.Personal.FindPersonal;
 import org.umssdiplo.automationv01.core.utils.CommonEvents;
 
 public class SubMenuPersonal extends BasePage {
 
-    @FindBy(xpath = "//button[@id='users']")
-    private WebElement usuarioSubMenu;
+    @FindBy(id = "users")
+    private WebElement subMenuUsuario;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: please change text to english. Ej.: subMenuUser

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 15]
```diff
@@ -4,19 +4,28 @@
...
+
+    @FindBy(id = "personal")
+    private WebElement subMenuPersonal;
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: please change text to english. Ej.: subMenuPerson

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 16]
```diff
@@ -4,19 +4,28 @@
...
 
     public SubMenuPersonal() {
-        CommonEvents.isVisible(usuarioSubMenu);
+        CommonEvents.isVisible(subMenuUsuario);
     }
 
     public ListUser selectSubMenuUsuario() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: please change text to english. Ej.: selectSubMenuUser

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java [hunk 17]
```diff
@@ -4,19 +4,28 @@
...
-        CommonEvents.clickButton(usuarioSubMenu);
+        CommonEvents.clickButton(subMenuUsuario);
         return new ListUser();
     }
+
+    public FindPersonal selectSubMenuPersonal() {
```
Comment (jdpapiero) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Menu/SubMenuPersonal.java]: please change text to english. Ej.: selectSubMenuPerson

File: src/test/resources/features/Personal/Personal.feature [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+    And seleccionar menu 'Personal' en la pagina 'Menu Principal'
+    And seleccionar submenu 'Personal' en menu 'Personal'
+    When ingresar <correo> para 'Buscar Personal' en la tabla
```
Comment (jdpapiero) [inline src/test/resources/features/Personal/Personal.feature]: rewrite better this step. Ej.: ingresar <correo> en el campo <Correo> de la pagina Personal para 'Buscar Personal' en la lista

**Review by jdpapiero — CHANGES_REQUESTED** [review]
review observations

**Review by jdpapiero — APPROVED** [review]
approve

**Review by marbusjim — APPROVED** [review]
approve

**Review by ivanMise — APPROVED** [review]

**Review by ctola91 — COMMENTED** [review]
Approved

**Pull request merged**